lib/ansible/modules/storage/netapp/na_ontap_net_routes.py

```diff
@@ -0,0 +1,245 @@
+
+    def delete_net_route(self):
+        """
+        Deletes a given Route
+        """
+        route_obj = netapp_utils.zapi.NaElement('net-routes-destroy')
+        route_obj.add_new_child("destination", self.destination)
+        route_obj.add_new_child("gateway", self.gateway)
+        try:
+            self.server.invoke_successfully(route_obj, True)
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error deleting net route: %s'
+                                  % (to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def modify_net_route(self):
+        """
+        Modify a net route
+        """
+        self.delete_net_route()
+        route_obj = netapp_utils.zapi.NaElement('net-routes-create')
+        if self.new_destination:
+            route_obj.add_new_child("destination", self.new_destination)
+        else:
+            route_obj.add_new_child("destination", self.destination)
+        if self.new_gateway:
+            route_obj.add_new_child("gateway", self.new_gateway)
+        else:
+            route_obj.add_new_child("gateway", self.gateway)
+        if self.new_metric:
+            route_obj.add_new_child("metric", self.new_metric)
+        else:
+            route_obj.add_new_child("metric", self.metric)
+        # if anything goes wrong creating the new net route,
+        # restore the old one
+        try:
+            self.server.invoke_successfully(route_obj, True)
+        except netapp_utils.zapi.NaApiError:
+            self.create_net_route()
+
+    def does_route_exists(self):
+        """
+        Checks to see if a route exist or not
+        :return: True if a route exists, False if it does not
+        """
+        route_obj = netapp_utils.zapi.NaElement('net-routes-get')
+        route_obj.add_new_child("destination", self.destination)
+        route_obj.add_new_child("gateway", self.gateway)
+        try:
+            self.server.invoke_successfully(route_obj, True)
+        except netapp_utils.zapi.NaApiError:
+            return False
+        return True
+
+    def apply(self):
+        """
+        Run Module based on play book
+        """
+        changed = False
+        netapp_utils.ems_log_event("na_ontap_net_routes", self.server)
+        route_exists = False
+        existing_route = self.does_route_exists()
+
+        if existing_route:
+            route_exists = True
+            if self.state == 'absent':
+                changed = True
+            elif self.state == 'present':
+                if self.new_destination or \
+                        self.new_gateway or self.new_metric:
+                    changed = True
+        else:
+            if self.state == 'present':
+                changed = True
+
+        if changed:
+            if self.module.check_mode:
+                pass
+            else:
+                if not route_exists:
```

Comment:
The "modification" case is not idempotent. If you were to run the modify task twice, the first time would modify as expected, the second time would re-create the original route. If `new_X` is specified, it should assume it's running in a modification mode- if the original value doesn't exist, ensure that the `new_X` version does, then return `changed: false`, else error (invalid route or something).

Reply:
We'll look into fixing this in the next release.